shiny/ui/_toast.py

```diff
@@ -0,0 +1,585 @@
+
+        # Build payload with flattened structure (no nested options)
+        payload: ToastPayload = {
+            "html": toasted["html"],
+            "deps": toasted["deps"],
+            "id": id,
+            "position": self.position,
+            "autohide": self.autohide,
+        }
+
+        # Add delay only if present
+        if self.duration is not None:
+            payload["duration"] = self.duration
+
+        return payload
+
+    def tagify(self, id: Optional[str] = None) -> Tag:
+        """Convert to HTML Tag object."""
+        # Determine accessibility attributes based on type
+        if self.type == "danger":
+            role = "alert"
+            aria_live = "assertive"
+        else:
+            role = "status"
+            aria_live = "polite"
+
+        # Start building classes
+        classes = ["toast"]
+        if self.type:
+            classes.append(f"text-bg-{self.type}")
+
+        # Build toast contents
+        contents: list[TagChild] = []
+
+        # Close button for header (if needed)
+        close_button = tags.button(
+            type="button",
+            class_="btn-close",
+            data_bs_dismiss="toast",
+            aria_label="Close",
+        )
+
+        # Build header if present
+        if self.header is not None:
+            # Check if header is already a ToastHeader or just text/tags
+            if isinstance(self.header, str):
+                # String header: convert to strong.me-auto
+                header_content = tags.strong(self.header, class_="me-auto")
+            elif isinstance(self.header, ToastHeader):
+                # ToastHeader: use tagify with closable flag
+                header_tag = self.header.tagify(closable=self.closable)
+                contents.append(header_tag)
+                header_content = None  # Already added
+            else:
+                # Otherwise pass through directly (assume user knows what they're doing)
+                header_content = self.header
+
+            # If we have header_content (not already added as ToastHeader), wrap it
+            if header_content is not None:
+                header_tag = div(
+                    {"class": "toast-header"},
+                    header_content,
+                    close_button if self.closable else None,
+                )
+                contents.append(header_tag)
+
+        # Build body with optional close button
+        # * If header exists, close button goes in header
+        # * If no header, close button goes in body (if closable)
+        body_has_close_btn = self.header is None and self.closable
+
+        if not body_has_close_btn and self.icon is None:
+            # Simple body: no close button, no icon
+            body_tag = div({"class": "toast-body"}, self.body)
+        else:
+            # Complex body: has close button and/or icon
+            body_contents: list[TagChild] = []
+
+            # Add icon if present
```

Comment:
Some of these comments smell of AI. Would be great to have a pass at reducing to only helpful ones